src/inet/IPAddress-StringFuncts.cpp

```diff
@@ -138,5 +139,33 @@ bool IPAddress::FromString(const char * str, size_t strLen, IPAddress & output)
     return res;
 }
 
+bool IPAddress::FromString(const char * str, IPAddress & addrOutput, class InterfaceId & ifaceOutput)
+{
+    char * addrStr       = const_cast<char *>(str);
+    char * addrPart      = nullptr;
+    char * scopePart     = nullptr;
+    char * strtokContext = nullptr;
+
+    addrPart = strtok_r(addrStr, "%", &strtokContext);
+    if (addrPart != nullptr)
+    {
+        scopePart = strtok_r(nullptr, "%", &strtokContext);
+    }
+
+    if (addrPart == nullptr || scopePart == nullptr)
+    {
+        return Inet::IPAddress::FromString(addrStr, addrOutput);
```

Comment:
Should set InterfaceId to something sane here, right?  Like the null interface id?

Reply:
Yeah good catch, done